web_src/js/features/repo-diff.js

```diff
@@ -89,6 +90,18 @@ export function initRepoDiffConversationNav() {
   });
 }
 
+// Will be called when the show more (files) button has been pressed
+function onShowMoreFiles() {
+  initRepoIssueContentHistory();
+  initViewedCheckboxListenerFor(document);
+
+  // The number of files is constant, but the number of viewed files can change because more files could be viewed
+  /* eslint-disable */
+  window.config.pageData.numberOfViewedFiles = document.querySelectorAll(`${viewedCheckboxSelector} > input[type=checkbox][checked]`).length;
+  /* eslint-enable */
```

Comment:
If anyone has an idea what eslint is complaining about here, I'm happy to modify it.
But to me, the complaint (`destructuring instead of properties`) does not make sense here.

Reply:
The problem is:

There was `const {csrfToken} = window.config;` above. So here `window.config.pageData` is wordy, it should be rewritten to:

 ```
const {csrfToken, pageData} = window.config;  // that's what `destructuring instead of properties` means
pageData.Xxx.Yyy = .....
```

And some other suggestions:

1. The usage of `pageData` had better to be scoped. It may provide data for different frontend modules, so it's better to have a `pageData['prReviewFile']` to contain all related variables.
2. `viewedCheckboxSelector` had better to be kept non-exported. For example, pass the data into the module function: `refreshViewedFilesSummary(pageData.prReviewFile)`